tests/syft/core/pointer/garbage_collection/inplace_op_gc_test.py

```diff
@@ -0,0 +1,38 @@
+# stdlib
+from typing import Any
+from typing import Tuple
+
+# third party
+import pytest
+import torch
+
+# syft absolute
+import syft as sy
+
+test_args = [
+    (sy.lib.python.List([1, 2, 3]), "append", (5,)),
+    (torch.Tensor([1, 2, 3]), "add_", (torch.Tensor([1, 2, 3]),)),
+    (sy.lib.python.collections.OrderedDict(), "__setitem__", ("key", "value")),
+]
+
+
+@pytest.mark.parametrize("obj, op, args", test_args)
+def test_inplace_ops_gc_test(
+    obj: Any,
+    op: str,
+    args: Tuple[Any],
+    node: sy.VirtualMachine,
+    client: sy.VirtualMachineClient,
+) -> None:
+
+    object_ptr = obj.send(client, pointable=False)
+    initial_id = object_ptr.id_at_location
+    assert len(node.store) == 1
+
+    func = getattr(object_ptr, op)
+    func(*args)  # inplace op
```

Comment:
We need to check the return pointer of the Tensor op because it returns a second pointer to the same value.